builder/dockerfile/builder.go

```diff
@@ -271,62 +274,97 @@ func emitImageID(aux *streamformatter.AuxFormatter, state *dispatchState) error
 	}
 	return aux.Emit(types.BuildResult{ID: state.imageID})
 }
+func convertMapToEnvs(m map[string]string) []string {
+	result := []string{}
+	for k, v := range m {
+		result = append(result, k+"="+v)
+	}
+	return result
+}
+func processMetaArg(meta instructions.ArgCommand, shlex *ShellLex, args *buildArgs) error {
+	envs := convertMapToEnvs(args.GetAllAllowed())
```

Comment:
Add comment that this is because `shlex` currently expects concatenated string format.